### ISSUE TITLE
Look into moving solution interpretation out from GeneralPhysics to the UI code.

### ISSUE THREAD

**Issue by andybalaam** [andybalaam/rabbit-escape#283]
As mentioned in #279 look into whether we can move the solution interpretation further out into the UI code, instead of embedding it in GeneralPhysics.

The reason why I feel this might be right is that it would mean we didn't need the UiPlayback interface and the tangled dependency cycle it represents.

**Comment by colonelfazackerley** [comment]
If android also gets demo mode, it makes sense to have as much as possible and in render or engine.

**Comment by andybalaam** [comment]
Yes, but something just feels wrong at the moment. 90% of the logic is shared already (in SolutionInterpreter) but I am concerned that the control is in the wrong place. I may well turn out to be wrong - I suggest you just ignore this until I find time to investigate and substantiate or throw out this feeling.
